projects/ccf-eui/src/app/app.component.ts

```diff
@@ -86,6 +90,16 @@ export class AppComponent implements OnInit {
       data: {preClose: () => {snackBar.dismiss();} },
       duration: this.tracking.snapshot.allowTelemetry === undefined ? Infinity : 3000
     });
+
+    // Sets initial theme according to user theme preference
+    if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
+      this.theming.setTheme(this.darkTheme);
+    }
+
+    // Listens for changes in user theme preference
+    window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', e => {
```

Comment:
You should also guard this statement like above, putting all in the `if (window.matchMedia)` block